### ISSUE TITLE
Distance - Receipt briefly shows placeholder after receipt is generated

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51785]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9-0-56-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+kh2310002@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace chat
3. Submit a manual expense (if there is no other expense)
4. Go offline
5. Submit a distance expense
6. Go to transaction thread
7. Click Distance
8. Go online
9. Wait for the receipt in the background to generate completely
10. Click outside the RHP to dismiss it

## Expected Result:
There will be no changes to the distance receipt and report header

## Actual Result:
The distance receipt shows placeholder thumbnil and header changes to "0.00 for Pending..." briefly

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/7aab32db-1702-4a61-9ac5-1eb42c9ccd89

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by NikkiWines** [comment]
This isn't breaking enough functionality to be considered a deploy blocker but it should be fixed.

**Comment by NikkiWines** [comment]
However, most likely coming from https://github.com/Expensify/App/pull/51519 cc: @Gonals @ikevin127 @wildan-m

**Comment by ikevin127** [comment]
Indeed, this is not a blocker and we discussed it in the PR (https://github.com/Expensify/App/pull/51519#discussion_r1817955794) and we decided to ship it like this because the fix for it was out of scope for the issue.

Therefore even though this is technically a regression of our PR, we decided not to count it as regression when it comes to compensation.

**Comment by NikkiWines** [comment]
Ok, marking this as a new issue then

**Comment by wildan-m** [comment]
@NikkiWines The issue is likely internal and should be addressed in the backend to resolve discrepancies between transactionBackup and transaction data during online updates. Please refer to this link for an initial investigation into the missing fields. https://github.com/Expensify/App/issues/48630#issuecomment-2438017630.

**Comment by sakluger** [comment]
This falls under `#quality` since the PR it stems from (https://github.com/Expensify/App/pull/51519#discussion_r1817955794) was also under `#quality`.

I'll mark as internal for now and raise a discussion in Slack around prioritization.

**Comment by sakluger** [comment]
No updates.

**Comment by sakluger** [comment]
No updates

**Comment by jacobkim9881** [comment]
This issue occurs, because the receipt data isn't added into `queuedOnyxUpdates` array. After clicking outside of RHP, the page is navigated from modal to distance request details and then `openReport` is activated to flush `queuedOnyxUpdates` array to update. So w/o the receipt data former data is updated.

**Comment by jacobkim9881** [comment]
@sakluger Sure. Since I haven't had an access to BE API, I think the internal team would take this issue.

**Comment by sakluger** [comment]
No updates. Still LOW priority.

**Comment by sakluger** [comment]
Still low priority.

**Comment by sakluger** [comment]
I asked again about the priority of this issue: https://expensify.slack.com/archives/C05LX9D6E07/p1738168389557229?thread_ts=1730486940.509219&cid=C05LX9D6E07

**Comment by sakluger** [comment]
We decided to close the issue.
